contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,227 @@
+            } else if (transferIndex < targetsLength) {
+                // Handle any remaining target amounts after the transfer process.
+                createProxyDelegatorAndTransfer(target, amount);
+            }
+        }
+
+        if (sourcesLength > 0) {
+            _burnBatch(msg.sender, sources, amounts[:sourcesLength]);
+        }
+        if (targetsLength > 0) {
+            _mintBatch(msg.sender, targets, amounts[:targetsLength], "");
+        }
+    }
+
+    /**
+     * @dev Processes the delegation transfer between a source delegate and a target delegate.
+     * @param source The source delegate from which tokens are being withdrawn.
+     * @param target The target delegate to which tokens are being transferred.
+     * @param amount The amount of tokens transferred between the source and target delegates.
+     */
+    function _processDelegation(
+        address source,
+        address target,
+        uint256 amount
+    ) internal {
+        uint256 balance = getBalanceForDelegate(source);
+
+        assert(amount <= balance);
+
+        deployProxyDelegatorIfNeeded(target);
+        transferBetweenDelegators(source, target, amount);
+
+        emit DelegationProcessed(source, target, amount);
+    }
+
+    /**
+     * @dev Reimburses any remaining source amounts back to the delegator after the delegation transfer process.
+     * @param source The source delegate from which tokens are being withdrawn.
+     * @param amount The amount of tokens to be withdrawn from the source delegate.
+     */
+    function _reimburse(address source, uint256 amount) internal {
+        // Transfer the remaining source amount or the full source amount
+        // (if no remaining amount) to the delegator
+        address proxyAddressFrom = retrieveProxyContractAddress(token, source);
+        token.transferFrom(proxyAddressFrom, msg.sender, amount);
+    }
+
+    function setUri(string memory uri) external onlyOwner {
+        _setURI(uri);
+    }
+
+    function createProxyDelegatorAndTransfer(
+        address target,
+        uint256 amount
+    ) internal {
+        address proxyAddress = deployProxyDelegatorIfNeeded(target);
+        token.transferFrom(msg.sender, proxyAddress, amount);
+    }
+
+    function transferBetweenDelegators(
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        address proxyAddressFrom = retrieveProxyContractAddress(token, from);
+        address proxyAddressTo = retrieveProxyContractAddress(token, to);
+        token.transferFrom(proxyAddressFrom, proxyAddressTo, amount);
+    }
+
+    function deployProxyDelegatorIfNeeded(
+        address delegate
+    ) internal returns (address) {
+        address proxyAddress = retrieveProxyContractAddress(token, delegate);
+
+        // check if the proxy contract has already been deployed
+        bytes memory bytecode;
+        assembly {
+            bytecode := extcodesize(proxyAddress)
```

Comment:
This returns the *size* of the bytecode, not the actual bytecode. I'm surprised this compiles!

Can you also check why there aren't any failing tests as a result of this?

Reply:
Right, but the variable is of type `bytes memory` and you're asking for its `length`. It should be a`uint`.